CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -259,19 +259,32 @@ protected static function flushCaches() {
       // Someone else is kindly doing the refresh for us right now.
       return;
     }
+
+    // Get the list of expired smart groups that may need flushing
     $params = [1 => [self::getCacheInvalidDateTime(), 'String']];
-    $groupsDAO = CRM_Core_DAO::executeQuery("SELECT id FROM civicrm_group WHERE cache_date <= %1", $params);
+    $groupsThatMayNeedToBeFlushedSQL = 'SELECT id FROM civicrm_group WHERE saved_search_id IS NOT NULL AND (cache_date <= %1 OR cache_date IS NULL)';
```

Comment:
this needs to be `(saved_search_id IS NOT NULL OR children <> '')`

I'm not sure if children would be NULL or an empty string but `<> ''` should get both

Reply:
Updated!